BROOKLYN-338: BrooklynApi fixes for karaf

aledsage opened the pull request:
These aren’t enough to fix BROOKLYN-338, but it’s a start:
* Ensures rest-client imports the resteasy marshallers package,
  which it calls with reflection (via the resteasy ProxyBuilder).
* Passes the ProxyBuilder a classloader that has access to the 
  rest-api classes and to rest-client’s classloader (which thus 
  can access the marshallers).

neykov commented on rest/rest-client/pom.xml:
```diff
@@ -144,5 +144,21 @@
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.5.3</version>
```
Remove version, already defined (as `2.5.4`) in `brooklyn-parent`.

neykov commented:
LGTM, build failure is unrelated.

aledsage commented:
Jenkins failure was for `OsTest.testDeleteRecursivelyEmptyDir`, failing for the 1000 attempts like:

    2016-08-23 22:38:29,719 WARN  Attempt to create temp dir failed /tmp/OsTest-IicA. Either an IO error (disk full, no rights) or someone else created the folder after the !exists() check.

Presumably "just" the build machine had run out of disk space! So assuming it wasn't our tests that filled up the disk, then this error is not related to our code.

---
I've incorporated Svet's comment and squashed the commits.

neykov commented:
Tests are passing locally, merging.
